CharlieBackend.Business/Services/CourseService.cs

```diff
@@ -97,5 +102,22 @@ public Task<bool> IsCourseNameTakenAsync(string courseName)
         {
             return _unitOfWork.CourseRepository.IsCourseNameTakenAsync(courseName);
         }
+
+        public async Task<Result<bool>> DisableCourceAsync(long id)
+        {
+            var studentGroup = await _unitOfWork.StudentGroupRepository.GetAllAsync();
```

Comment:
You're getting all the groups from db here. Please use _unitOfWork.StudentGroupRepository.GetQueryableNoTracking().AnyAsync